emails/tests/views_tests.py

```diff
@@ -1270,36 +1315,406 @@ def test_complaint_disables_mask(self):
             "complaint_subtype": None,
             "complaint_user_agent": "ExampleCorp Feedback Loop (V0.01)",
             "domain": "test.com",
-            "relay_action": "auto_block_spam",
+            "relay_action": "disable_mask",
             "user_match": "found",
+            "mask_match": "found",
             "fxa_id": self.sa.uid,
+            "found_in": "all",
         }
 
     @override_flag("developer_mode", active=True)
-    def test_complaint_mpp_3932(self):
-        """MPP-3932: Log notification for all complaints for developer_mode users."""
+    def test_complaint_developer_mode(self):
+        """Log complaint notification for developer_mode users."""
+
+        simulator_complaint_message = deepcopy(self.complaint_msg)
+        simulator_complaint_message["complaint"]["complainedRecipients"] = [
+            {"emailAddress": f"complaint+{self.ra.address}@simulator.amazonses.com"}
+        ]
+        complaint_body = {"Message": json.dumps(simulator_complaint_message)}
+
         with self.assertLogs(INFO_LOG) as logs:
-            response = _sns_notification(self.complaint_body)
+            response = _sns_notification(complaint_body)
         assert response.status_code == 200
 
         self.user.profile.refresh_from_db()
         assert self.user.profile.auto_block_spam is True
         self.mock_ses_client.send_raw_email.assert_not_called()
 
-        assert len(logs.records) == 2
-        record_mpp_3932 = logs.records[0]
-        assert record_mpp_3932.msg == "_handle_complaint: MPP-3932"
-        assert getattr(record_mpp_3932, "mask_id") == "unknown"
-        assert getattr(record_mpp_3932, "dev_action") == "log"
-        assert getattr(record_mpp_3932, "parts") == 1
-        assert getattr(record_mpp_3932, "part") == 0
-        notification_gza85 = getattr(record_mpp_3932, "notification_gza85")
+        (rec1, rec2) = logs.records
+        assert rec1.msg == "_handle_complaint: developer_mode"
+        assert getattr(rec1, "mask_id") == self.ra.metrics_id
+        assert getattr(rec1, "dev_action") == "log"
+        assert getattr(rec1, "parts") == 1
+        assert getattr(rec1, "part") == 0
+        notification_gza85 = getattr(rec1, "notification_gza85")
         log_complaint = decode_dict_gza85(notification_gza85)
-        expected_log_complaint = json.loads(self.complaint_body["Message"])
-        assert log_complaint == expected_log_complaint
+        assert log_complaint == simulator_complaint_message
 
-        record = logs.records[1]
-        assert record.msg == "complaint_notification"
+        assert rec2.msg == "complaint_notification"
+
+    @override_flag("developer_mode", active=True)
+    def test_complaint_developer_mode_missing_mask(self):
+        """If the simulator email can not be turned into a mask, it is not changed."""
```

Comment:
sugguestion (non-blocking): add a little more detail here? I don't understand what "simulator email can not be turned into a mask" means.

Reply:
I'm going to move these "cover all the edge cases" tests to a new class, so `ComplaintHandlingTest` is more like integration tests.